src/staticanalysis/bot/static_analysis_bot/infer/infer.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import subprocess
+
+from cli_common.log import get_logger
+from static_analysis_bot import Issue
+from static_analysis_bot import stats
+from static_analysis_bot.config import settings
+from static_analysis_bot.revisions import Revision
+
+logger = get_logger(__name__)
+
+ISSUE_MARKDOWN = '''
+## infer {type}
+
+- **Message**: {message}
+- **Location**: {location}
+- **In patch**: {in_patch}
+- **Clang check**: {check}
+- **Publishable check**: {publishable_check}
+- **Third Party**: {third_party}
+- **Expanded Macro**: {expanded_macro}
+- **Publishable **: {publishable}
+- **Is new**: {is_new}
+
+```
+{body}
+```
+'''
+
+INFER_SETUP_CMD = [
+    'gecko-env',
+    './mach', 'artifact', 'toolchain',
+    '--from-build', 'linux64-infer'
+]
+
+
+class Infer(object):
+    '''
+    Infer runner
+    '''
+    def __init__(self, validate_checks=True):
+        self.binary = os.path.join(
+            os.environ['MOZBUILD_STATE_PATH'],
+            'infer', 'infer', 'bin', 'infer',
+        )
+        assert os.path.exists(self.binary), \
+            'Missing infer in {}'.format(self.binary)
+
+    @stats.api.timed('runtime.infer')
+    def run(self, revision):
+        '''
+        Run modified files with specified checks through infer
+        using threaded workers (communicate through queues)
+        Output a list of InferIssue
+        '''
+        assert isinstance(revision, Revision)
+        self.revision = revision
+
+        # Run all files in a single command
+        # through mach static-analysis
+        cmd = [
+            'gecko-env',
+            './mach', '--log-no-times', 'static-analysis', 'check-java'
+        ] + list(revision.files)
+        logger.info('Running static-analysis', cmd=' '.join(cmd))
+
+        # Run command
+        try:
+            infer_output = subprocess.check_output(cmd, cwd=settings.repo_dir)
+        except subprocess.CalledProcessError as e:
+            logger.error('Mach static analysis failed: {}'.format(e.output))
+            raise
+
+        import json
+        report_file = os.path.join(settings.repo_dir, 'infer-out', 'report.json')
+        infer_output = json.load(open(report_file))
+        # Dump raw clang-tidy output as a Taskcluster artifact (for debugging)
```

Comment:
Please add a newline before a comment. That's just some nitpicking, but i find it way more readable